### ISSUE TITLE
`no-array-for-each` invalid fix

### ISSUE THREAD

**Issue by pustovitDmytro** [sindresorhus/eslint-plugin-unicorn#1318]
Consider foreach calls some logic as return side-effect:
```js
function handleTwo() {}
function handleNotTwo() {}

[ 1, 2, 3, 4 ].forEach(item => {
    if (item === 2) return handleTwo();
    handleNotTwo();
});
```

after fixing by ```no-array-for-each``` next snippet is generated:

```js
function handleTwo() {}
function handleNotTwo() {}

for (const item of [ 1, 2, 3, 4 ]) {
    if (item === 2)  handleTwo(); continue;
    handleNotTwo();
}
```

here ```handleTwo``` is being called in case item equals 2 as expected, but then continue is called out of if scope.
as a result  ```handleNotTwo```  becomes  unreachable, and snippet **works in different way** then initial  forEach.

Consider addding scope here should work as expected:

```return handleTwo();``` => ```{ handleTwo(); continue; }```

so snippet becomes:
```js
function handleTwo() {}
function handleNotTwo() {}

for (const item of [ 1, 2, 3, 4 ]) {
    if (item === 2)  { handleTwo(); continue; }
    handleNotTwo();
}
```

**Comment by fisker** [comment]
Thanks for reporting this.
